lib/util/pssh.js

```diff
@@ -189,4 +208,43 @@ shaka.util.Pssh = class {
 
     return shaka.util.Uint8ArrayUtils.concat(...dedupedInitDatas);
   }
+
+  /**
+   * Get a pssh from the given keySystem and moov.
+   *
+   * @param {!Uint8Array} initData
+   * @param {string} keySystem
+   * @return {!Uint8Array}
+   */
+  static getPsshFromInitData(initData, keySystem) {
+    const systemId = shaka.util.Pssh.defaultSystemIds_.get(keySystem);
+    const psshInfo = new shaka.util.Pssh(initData, true);
+
+    // If there is only a single pssh, return the original array.
```

Comment:
This isn't right when you've extracted a single pssh from inside a moov atom.  I think this special case should be removed.

Reply:
Done! I remove it.